pjmedia/src/pjmedia-audiodev/alsa_dev.c

```diff
@@ -699,6 +702,9 @@ static pj_status_t open_playback (struct alsa_stream* stream,
 	tmp_buf_size = (rate / 1000) * param->output_latency_ms;
     else
 	tmp_buf_size = (rate / 1000) * PJMEDIA_SND_DEFAULT_PLAY_LATENCY;
+    if (tmp_buf_size < tmp_period_size * 2)
+        tmp_buf_size = tmp_period_size * 2;
+    tmp_buf_size *= param->channel_count;
```

Comment:
In my opinion this is **not** correct.

The function `snd_pcm_hw_params_set_buffer_size_near()` expects the buffer size in frames. In ALSA terms a frame is the sum of all samples for on given point in time. e.g. For a stereo device 1 frame would consist of 2 samples. [This article](https://www.linuxjournal.com/article/6735) gives a good overview.

Because `snd_pcm_hw_params_set_buffer_size_near()` expects the number of frames as argument, there's no point in multiplying the buffer size with the channel count. The effect will be an increased buffer latency, dependent on the channel count, which is obviously not on purpose.

The same applies to line 833.

Reply:
You're right, we missed it. The article: https://www.alsa-project.org/main/index.php/FramesPeriods mentioned by @nanangizz above actually also says the same thing.
"1 frame represents 1 analog sample from all channels".

Thanks for noticing. Created PR #2549 for this.

Reply:
Good catch! Thanks @mike2307.

Btw, the new PR ID should be #2556?